internal/capability/operator_install.go

```diff
@@ -49,77 +57,92 @@ func OperatorInstall(s operator.SubscriptionData, c operator.Client) error {
-			Name:             operatorGroup,
+			Name:             m.operatorGroup,
 			TargetNamespaces: []string{},
 		}
-		c.CreateOperatorGroup(context.Background(), opGroupData, namespace)
+		c.CreateOperatorGroup(context.Background(), opGroupData, m.namespace)
 
 	case operatorv1alpha1.InstallModeTypeSingleNamespace:
 
-		operator.CreateNamespace(context.Background(), targetNs1)
+		operator.CreateNamespace(context.Background(), m.targetNs1)
 		opGroupData := operator.OperatorGroupData{
-			Name:             operatorGroup,
-			TargetNamespaces: []string{targetNs1},
+			Name:             m.operatorGroup,
+			TargetNamespaces: []string{m.targetNs1},
 		}
-		installedNS = append(installedNS, targetNs1)
-		c.CreateOperatorGroup(context.Background(), opGroupData, namespace)
+		m.installedNS = append(m.installedNS, m.targetNs1)
+		c.CreateOperatorGroup(context.Background(), opGroupData, m.namespace)
 
 	case operatorv1alpha1.InstallModeTypeOwnNamespace:
 		opGroupData := operator.OperatorGroupData{
-			Name:             operatorGroup,
-			TargetNamespaces: []string{namespace},
+			Name:             m.operatorGroup,
+			TargetNamespaces: []string{m.namespace},
 		}
-		c.CreateOperatorGroup(context.Background(), opGroupData, namespace)
+		c.CreateOperatorGroup(context.Background(), opGroupData, m.namespace)
 
 	case operatorv1alpha1.InstallModeTypeMultiNamespace:
 
-		operator.CreateNamespace(context.Background(), targetNs1)
-		operator.CreateNamespace(context.Background(), targetNs2)
+		operator.CreateNamespace(context.Background(), m.targetNs1)
+		operator.CreateNamespace(context.Background(), m.targetNs2)
 		opGroupData := operator.OperatorGroupData{
-			Name:             operatorGroup,
-			TargetNamespaces: []string{targetNs1, targetNs2},
+			Name:             m.operatorGroup,
+			TargetNamespaces: []string{m.targetNs1, m.targetNs2},
 		}
-		installedNS = append(installedNS, targetNs1)
-		installedNS = append(installedNS, targetNs2)
-		c.CreateOperatorGroup(context.Background(), opGroupData, namespace)
+		m.installedNS = append(m.installedNS, m.targetNs1)
+		m.installedNS = append(m.installedNS, m.targetNs2)
+		c.CreateOperatorGroup(context.Background(), opGroupData, m.namespace)
 
 	}
 
-	// create subscription per operator package/channel
-	sub, err := c.CreateSubscription(context.Background(), s, namespace)
-	if err != nil {
-		logger.Debugf("Error creating subscriptions: %w", err)
-		return err
-	}
+}
 
-	if err = c.WaitForInstallPlan(context.Background(), sub); err != nil {
-		logger.Debugf("Waiting for InstallPlan: %w", err)
-		return err
-	}
-	// check/approve install plan
-	// TODO: check the name standard for installPlan
-	err = c.InstallPlanApprove(namespace)
-	if err != nil {
-		logger.Debugf("Error creating subscriptions: %w", err)
-		return err
-	}
+func cleanUp(s operator.SubscriptionData, c operator.Client, m operatorData) {
 
-	csvStatus, err := c.WaitForCsvOnNamespace(namespace)
+	csvStatus, err := c.WaitForCsvOnNamespace(m.namespace)
```

Comment:
Just a small nit, this is not part of the clean-up process and can go back to the OperatorInstallAllFromCatalog function.